runtime_scan/pkg/orchestrator/scanwatcher/watcher.go

```diff
@@ -0,0 +1,220 @@
+// Copyright © 2023 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scanwatcher
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/openclarity/vmclarity/api/models"
+	"github.com/openclarity/vmclarity/runtime_scan/pkg/orchestrator/common"
+	runtimeScanUtils "github.com/openclarity/vmclarity/runtime_scan/pkg/utils"
+	"github.com/openclarity/vmclarity/shared/pkg/backendclient"
+)
+
+const (
+	DefaultPollInterval     = time.Minute
+	DefaultReconcileTimeout = time.Minute
+)
+
+type ScanReconcileEvent struct {
+	ScanID models.ScanID
+}
+
+type (
+	ScanQueue      = common.Queue[ScanReconcileEvent]
+	ScanPoller     = common.Poller[ScanReconcileEvent]
+	ScanReconciler = common.Reconciler[ScanReconcileEvent]
+)
+
+type Config struct {
+	Backend          *backendclient.BackendClient
+	PollPeriod       time.Duration
+	ReconcileTimeout time.Duration
+}
+
+func New(c Config) *Watcher {
+	logger := log.WithFields(log.Fields{"controller": "ScanWatcher"})
+	return &Watcher{
+		logger,
+		c.Backend,
+		c.PollPeriod,
+		c.ReconcileTimeout,
+	}
+}
+
+type Watcher struct {
+	logger           *log.Entry
+	client           *backendclient.BackendClient
+	pollPeriod       time.Duration
+	reconcileTimeout time.Duration
+}
+
+func (w *Watcher) Start(ctx context.Context) {
+	queue := common.NewQueue[ScanReconcileEvent]()
+
+	poller := &ScanPoller{
+		Logger:     w.logger,
+		PollPeriod: w.pollPeriod,
+		Queue:      queue,
+		GetItems:   w.GetAbortedScans,
+	}
+	poller.Start(ctx)
+
+	reconciler := &ScanReconciler{
+		ReconcileTimeout:  w.reconcileTimeout,
+		Queue:             queue,
+		ReconcileFunction: w.Reconcile,
+	}
+	reconciler.Start(ctx)
+}
+
+func (w *Watcher) GetAbortedScans(ctx context.Context) ([]ScanReconcileEvent, error) {
+	scans, err := w.getScansByState(ctx, models.ScanStateAborted)
+	if err != nil || scans.Items == nil || len(*scans.Items) <= 0 {
+		return nil, err
```

Comment:
please wrap the error, in case `err==nil`

Reply:
Could you please elaborate on this? If `err != nil` returned by `getScansByState()` than `GetAbortedScans()` returnes error as well, otherwise not.